dipy/direction/bingham.py

```diff
@@ -0,0 +1,772 @@
+r"""Tools for fitting Bingham distributions to orientation distribution
+functions (ODF), as described in Riffert et al [1]_. The resulting
+distributions can further be used to compute ODF-lobe-specific measures such as
+the fiber density (FD) and fiber spread (FS) [1]_ and the orientation
+dispersion index (ODI) [2]_.
+
+References
+----------
+.. [1] Riffert TW, Schreiber J, Anwander A, Knösche TR. Beyond fractional
+        anisotropy: Extraction of bundle-specific structural metrics from
+        crossing fiber models. NeuroImage. 2014 Oct 15;100:176-91.
+.. [2] R. Neto Henriques, “Advanced methods for diffusion MRI data analysis
+        and their application to the healthy ageing brain.” Apollo -
+        University of Cambridge Repository, 2018. doi: 10.17863/CAM.29356.
+"""
+
+from math import cos, radians
+from warnings import warn
+import numpy as np
+
+from dipy.direction import peak_directions
+from dipy.reconst.shm import sh_to_sf
+from dipy.core.ndindex import ndindex
+from dipy.core.onetime import auto_attr
+
+
+def _bingham_fit_peak(sf, peak, sphere, max_angle):
+    r"""
+    Fit Bingham function on the ODF lobe aligned with peak.
+
+    Parameters
+    ----------
+    odf: 1d ndarray
+        The odf function evaluated on the vertices of `sphere`
+    peak: ndarray (3, 1)
+        The peak direction of the lobe to fit.
+    sphere: `Sphere` class instance
+        The Sphere providing the odf's discrete directions
+    max_angle: float
+        The maximum angle in degrees of the neighbourhood around
+        peak to consider for fitting.
+
+    Returns
+    -------
+    f0: float
+        Maximum amplitude of the distribution/peak.
+    k1: tuple of floats
+        Concentration parameter of major axis k1.
+    k2: tuple of floats
+        Concentration parameter of minor axis k2.
+    mu0: ndarray (3,) of floats
+        Main axis of ODF peak.
+    mu1: ndarray (3,) of floats
+        Major concentration axis.
+    mu2: ndarray (3,) of floats
+        Minor concentration axis.
+    """
+    # abs for twice the number of pts to fit
+    dot_prod = np.abs(sphere.vertices.dot(peak))
+    min_dot = cos(radians(max_angle))
```

Comment:
to be consistent with all the codebase, I would recommend using numpy here (`np.cos`).  You use it later your code, so any reason to use the math module here ?

Reply:
Legacy code from scilpy. No there is no reason for not using numpy, I changed it.